fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -682,6 +683,9 @@ public PlanFragment visitPhysicalOlapScan(PhysicalOlapScan olapScan, PlanTransla
                 )
         );
         olapScanNode.setPushDownAggNoGrouping(context.getRelationPushAggOp(olapScan.getRelationId()));
+        if (olapScan.getMutableState(MutableState.KEY_TOPNRF_ID).isPresent()) {
```

Comment:
we'd better not use mutable state to save this info, since it is very easy be borken by other rules after TopNScanOpt.